skimage/filters/_fft_based.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import scipy.fft as fft
+from skimage.exposure import rescale_intensity
+from skimage import dtype_limits
+
+
+def _get_ND_butterworth_filter(shape, factor, order, high_pass):
+    """
+    Get a N-dimensional Butterworth mask
+
+    Parameters
+    ----------
+    shape : tuple of ints
+        dimensions of the n-dimensional image and mask
+    factor : float
+        fraction of width and height of the filter where the cutoff should be
+    order : float
+        controls the slope in the cutoff region
+    high_pass : bool
+        whether the filter is high pass (low frequencies attenuated) or
+        low pass (high frequencies are attenuated)
+
+    Returns
+    -------
+    wfilt : numpy array of shape
+        The filter/fft mask
+    """
+    ranges = []
+    for d in shape:
+        # start and stop ensures center of mask aligns with center of fft
+        axis = np.arange(-(d - 1) // 2, (d - 1) // 2 + 1) / (d * factor)
+        ranges.append(axis**2)
+    # q2 = euclidian distance squared grid
+    q2 = np.sum(np.meshgrid(*ranges, indexing="ij"), axis=0)
```

Comment:
```suggestion
    q2 = functools.reduce(
        np.add, np.meshgrid(*ranges, indexing='ij', sparse=True)
    )
```
This is a more efficient way to do the same thing (in terms of both memory usage and computation). This way all arrays returned by meshgrid are size 1 on all but 1 axis and no large array with `len(shape + 1)` dimensions has to be created.

Reply:
This is a cool and elegant trick, and indeed with sparse we won't run into any memory issues. Thanks!